### PR TITLE
Pull request for Chapter 1 Problem set

### PR DESCRIPTION
**Pull request by OverclockedTim**
A problem set for Chapter 1. This initial checkin includes 8 problems for Chapter 1 that engaged readers can use to enhance and solidify their understanding of Chapter 1 topics. I believe in its current form, this problem set is helpful but that it also will need to be expanded in the future. If this pull request is well received, I hope to create similar problem sets for each of the remaining chapters.

### CODE REVIEW

**Comment by zackchase** [comment]
Hi Tim! Just graduated from PhD. Now can focus again :). Can you please copy-edit, then I'll merge. I'll do a code review to indicate some of the typos.

File: chapter01_crashcourse/chapter-one-problem-set.ipynb [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problems using ndarray [(Official Documentation)](https://mxnet.incubator.apache.org/api/python/ndarray.html) \n",
+    "\n",
+    "\n",
+    "Problem 1: Initialize an ndarray of dimension 1x256 on the GPU without overwriting its memory. Then, find the index of the maximum value in the array (argmax)"
```
Comment (zackchase) [inline chapter01_crashcourse/chapter-one-problem-set.ipynb]: NDArray

Comment (zackchase) [reply chapter01_crashcourse/chapter-one-problem-set.ipynb]: find the index *corresponding to* the maximum value in the array

File: chapter01_crashcourse/chapter-one-problem-set.ipynb [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    "# Problem 1 Work Area"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problems from Linear Algebra\n",
+    "\n",
+    "Problem 2: Create a 4x4 matrix of random value. Then create an 4x4. identity matrix (an identity of size n is the n × n square matrix with ones on the main diagonal and zeros elsewhere). Multiple the two together and verify that you get the original matrix back."
```
Comment (zackchase) [inline chapter01_crashcourse/chapter-one-problem-set.ipynb]: of random value*s*. -- I'd also be more specific, there are many ways to initialize random values. maybe uniformly random on the interval [0,1]?

Comment (zackchase) [reply chapter01_crashcourse/chapter-one-problem-set.ipynb]: "multiple the two" -> multiply the two

File: chapter01_crashcourse/chapter-one-problem-set.ipynb [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+   "outputs": [],
+   "source": [
+    "# Problem 2 Work Area"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Problem 3: Create a 3x3x20 tensor such that at every x,y coordinate, moving through the z coordinate list the [Fibonacci sequence](https://en.wikipedia.org/wiki/Fibonacci_number). So, at a z position of 0, the 3x3 matrix will be all 1s. At z-position 1, the 3x3 matrix will be all 1s. At z-position 2, the 3x3 matrix will be all 2s, at z-position 3, the 3x3 matrix will be all 3s and so forth.\n",
```
Comment (zackchase) [inline chapter01_crashcourse/chapter-one-problem-set.ipynb]: "moving through the z coordinate list the", -> list*s*

File: chapter01_crashcourse/chapter-one-problem-set.ipynb [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+   "outputs": [],
+   "source": [
+    "# Problem 4 Work Area"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Problem 5: Create a vector (0,1), and another vector (1,0), and use mxnet to calculate the angle between them. Remember that the dot product of two vectors is equal to the cossine of the angle between the vectors, and that the arccos function is the inverse of cosine."
```
Comment (zackchase) [inline chapter01_crashcourse/chapter-one-problem-set.ipynb]: (0,1) is weird notation, it looks like a shape tuple

File: chapter01_crashcourse/chapter-one-problem-set.ipynb [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    "# Problem 5 Work Area"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problems from Probability\n",
+    "\n",
+    "Problem 6: In the classic game of Risk, the attacker can roll a maximum of three dice, while the defender can roll a maximum of two dice. Simulate the attacking and defending dice using sample_multinomial to try to estimate the odds that an attacker will win against a defender when both are rolling the maximum number of dice."
```
Comment (zackchase) [inline chapter01_crashcourse/chapter-one-problem-set.ipynb]: put code - like the function name `sample_multinomial`, in backticks

**Comment by zackchase** [comment]
Any updates?

**Comment by OverclockedTim** [comment]
Hi Zack, I believe this should address all of the issues you raised - really sorry about the long time lag. I kept being sure I could get to it the following weekend and then having to deprioritize for one reason or another - hope this helps!

**Comment by zackchase** [comment]
Thanks!

**Pull request merged**